modules/core/src/lib/picking/pick-info.js

```diff
@@ -141,11 +140,20 @@ export function getLayerPickingInfo({layer, info, mode}) {
 }
 
 // Indentifies which viewport, if any corresponds to x and y
+// If multiple viewports contain the target pixel, use priority:
+//  - last viewport that the picked layer was rendered in
+//  - last viewport drawn
 // Returns first viewport if no match
-// TODO - need to determine which viewport we are in
-// TODO - document concept of "primary viewport" that matches all coords?
-// TODO - static method on Viewport class?
-function getViewportFromCoordinates({viewports}) {
-  const viewport = viewports[0];
-  return viewport;
+function getViewportFromCoordinates(viewports, pixel, layer) {
+  const lastDrawnViewport = layer && layer.internalState.viewport;
+  if (lastDrawnViewport && lastDrawnViewport.containsPixel(pixel)) {
+    return lastDrawnViewport;
+  }
+  // find the last viewport that contains the pixel
+  for (let i = viewports.length - 1; i >= 0; i--) {
+    if (viewports[i].containsPixel(pixel)) {
+      return viewports[i];
```

Comment:
I suppose there could also be a check if the picked layer was actually drawn by the viewport (layerFilters etc)? But since that wouldn't actually be a full solution, it may not matter.

Reply:
Yes that could be an alternative to using `lastDrawnViewport`